bot/cogs/nasa.py

```diff
@@ -0,0 +1,167 @@
+            )
+            embed.set_image(url=item["links"][0]["href"])
+            embed.set_footer(
+                text=f"ID: {item['data'][0]['nasa_id']} | Powered by HotWired"
+            )
+
+            await ctx.send(embed=embed)
+        else:
+            await ctx.send("No results found!")
+
+    @command(aliases=["nid"])
+    async def nasa_id(self, ctx: Context, id: str) -> None:
+        """Search for a picture on Nasa with nasa id."""
+        async with self.session.get(f"https://images-api.nasa.gov/asset/{id}") as resp:
+            data = await resp.json()
+
+        try:
+            items = data["collection"]["items"][0]["href"]
+            embed = Embed(
+                color=Color.blurple()
+            )
+            embed.set_image(url=items)
+            embed.set_footer(
+                text=f"ID: {id} | Powered by HotWired"
+            )
+        except KeyError:
+            await ctx.send("No results found!")
+
+        await ctx.send(embed=embed)
+
+    @command(aliases=["npatent"])
+    async def nasa_patent(self, ctx: Context, *, patent: str) -> None:
+        """Search for a picture on Nasa with nasa id."""
+        async with self.session.get(f"https://api.nasa.gov/techtransfer/patent/?{patent}&api_key={NASA_API}") as resp:
+            data = await resp.json()
+
+        items = data["results"]
+        if len(items) > 0:
+            rand_item = random.randint(0, len(items) - 1)
+            item = items[rand_item]
+
+            if len(remove_tags(item[3][:2000])) > 2048:
+                description = f"{item[3][:2000][:2045].strip()}..."
+            else:
+                description = item[3][:2000]
+
+            embed = Embed(
+                title=remove_tags(item[2]),
+                description=remove_tags(description),
+                color=Color.blurple()
+            )
+            embed.set_footer(
+                text="Powered by HotWired"
+            )
+
+            await ctx.send(embed=embed)
+        else:
+            await ctx.send("No results Found!")
+
+    @command()
+    async def epic(self, ctx: Context, max: int = 1) -> None:
+        """Get images from DSCOVR's Earth Polychromatic Imaging Camera. can specify a maximum number of images to retrieve."""
+        async with self.session.get("https://epic.gsfc.nasa.gov/api/images.php") as response:
+            json = await response.json()
+
+            for i in range(min(max, len(json))):
+                embed = Embed(
+                    ittle="EPIC image",
+                    description=json[i].get("caption"),
+                    colour=Color.blurple()
+                )
+                embed.set_image(url="https://epic.gsfc.nasa.gov/epic-archive/jpg/" + json[i]["image"] + ".jpg")
+
+                await ctx.send(embed=embed)
+
+    @command()
+    async def mars(self, ctx: Context, date: str, rover: str = random.choice(("curiosity", "opportunity", "spirit")), n: int = 1) -> None:
```

Comment:
Using `rover: str = random.choice((...))` won't do what you intend to do.

Function in definitions of another function will only get evaluated once, meaning when you start the program, this function will get evaluated and the `random.choice` will return one of the provided strings, after that it will always default to this pre-evaluated string instead of evaluating the choice every time the function runs. 

What you want to do instead is set `rover: t.Optional[str] = None` with a condition of
```py
if rover is None:
    rover = random.choice((...))
```

Note that you might need to move `rover` behind the `n` argument in the function as it will need to be specified if the user intends to change `n` a workaround to it would be something like this: ` t.Optional[t.Literal["curiosity", "opportunity", "spirit"]]` but that will limit the user to only using lowercase name for the rover names.